File: tests/leetcode/x/cpp/0009.cpp
#include <iostream>
using namespace std;

static bool isPalindrome(int x) {
    if (x < 0) return false;
    int original = x;
    long long rev = 0;
    while (x > 0) {
        rev = rev * 10 + (x % 10);
        x /= 10;
    }
    return rev == original;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int t;
    if (!(cin >> t)) return 0;
    while (t--) {
        int x;
        cin >> x;
        cout << (isPalindrome(x) ? "true" : "false") << "\n";
    }
    return 0;
}

File: tests/leetcode/x/csharp/0009.cs
using System;

public static class Program
{
    static bool IsPalindrome(int x)
    {
        if (x < 0) return false;
        int original = x;
        long rev = 0;
        while (x > 0)
        {
            rev = rev * 10 + (x % 10);
            x /= 10;
        }
        return rev == original;
    }

    public static void Main()
    {
        string input = Console.In.ReadToEnd();
        string[] parts = input.Split((char[])null, StringSplitOptions.RemoveEmptyEntries);
        if (parts.Length == 0) return;
        int idx = 0;
        int t = int.Parse(parts[idx++]);
        for (int i = 0; i < t; i++)
        {
            int x = int.Parse(parts[idx++]);
            Console.WriteLine(IsPalindrome(x) ? "true" : "false");
        }
    }
}
